src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java

```diff
@@ -38,7 +38,7 @@
 	public Link(String id, String anchor, int start, int end) {
 		super();
 		try {
-			this.id = java.net.URLDecoder.decode(id, "UTF-8");
+			this.id = java.net.URLDecoder.decode(id.replace("+", "%2B"), "UTF-8");
```

Comment:
We made a hack: try to decode, and if it fails just keep the un-decoded string.
Now it blows up with the "+" and we try to patch the hack.
I think we should try another approach or it will blow up again.
Is there a way to know if we need to apply decode?